src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/SyntaxListExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    public static class SyntaxListExtensions
+    {
+        internal static SyntaxNode PreviousSiblingOrSelf(this SyntaxList<RazorSyntaxNode> syntaxList, RazorSyntaxNode syntaxNode)
+        {
+            var index = syntaxList.IndexOf(syntaxNode);
+
+            if (index == 0)
```

Comment:
Should we add a `-1` protection case for the scenario where someone stores a syntax list and uses the wrong node with it? Ditto with below